papermill/execute.py

```diff
@@ -204,22 +203,29 @@ def _parameterize_notebook(nb, kernel_name, parameters):
     # Generate parameter content based on the kernal_name
     kernel_name = kernel_name or nb.metadata.kernelspec.name
     param_content = _build_parameter_code(kernel_name, parameters)
-
     param_cell_index = _find_parameters_index(nb)
-    if param_cell_index >= 0:
-        old_parameters = nb.cells[param_cell_index]
-        old_parameters.metadata['tags'].remove('parameters')
-        old_parameters.metadata['tags'].append('default parameters')
-
-    newcell = nbformat.v4.new_code_cell(source=param_content)
-    newcell.metadata['tags'] = ['parameters']
-
-    before = nb.cells[:param_cell_index + 1]
-    after = nb.cells[param_cell_index + 1:]
-    nb.cells = before + [newcell] + after
+    param_cell = _get_cell(nb, param_cell_index)
+    meta_param_cell_index = nb.metadata.papermill.get('parameter_cell', param_cell_index)
+
+    # Check if we have a prior run's data to override
+    if 'parameters' in nb.metadata.papermill and \
+            param_cell_index == meta_param_cell_index and \
+            param_cell:
```

Comment:
This seems like a dangerous comparison since it relies on ordering. Would it help if we assigned unique IDs to the cells in metadata?

Reply:
Possibly, or we can just not check param cell index at all? My original thinking was that if a user copied an executed notebook and changed it you'd want to treat it as a new notebook.

Actually a better way to achieve this would be to md5 the artifact and save that as a metadata marker marker. Then if a user makes changes the hash would differ.

Reply:
That seems sensible, though I haven't fully thought through this.